src/Control/Monad/Morph.hs

```diff
@@ -116,6 +120,54 @@ class MFunctor t where
     -}
     hoist :: (Monad m) => (forall a . m a -> n a) -> t m b -> t n b
 
+{-| Convert a Monad to an isomorphic transformation of Identity
+> mtrans (a >> b)   = mtrans a >> mtrans b
+-}
+class Monad m => MTrans t m where
+    mtrans :: m a -> t Identity a
+
+instance MTrans I.IdentityT Identity where
+    mtrans = I.IdentityT
+
+instance MTrans M.MaybeT Maybe where
+    mtrans = M.MaybeT <$> pure
+
+instance MTrans M.MaybeT (M.MaybeT Identity) where
+    mtrans = id
+
+instance MTrans (Ex.ExceptT e) (Either e) where
+    mtrans = Ex.except
+
+instance MTrans (Ex.ExceptT e) (Ex.Except e) where
+    mtrans = id
+
+instance (Monoid w) => MTrans (RWS.RWST r w s) (RWS.RWS r w s) where
+    mtrans = id
+
+instance (Monoid w) => MTrans (RWS'.RWST r w s) (RWS'.RWS r w s) where
+    mtrans = id
+
+instance MTrans (R.ReaderT r) (R.Reader r) where
+    mtrans = id
+
+instance MTrans (R.ReaderT r) ((->) r) where
+    mtrans = R.reader
+
+instance MTrans (S.StateT s) (S.State s) where
+    mtrans = id
+
+instance MTrans (S'.StateT s) (S'.State s) where
+    mtrans = id
+
+instance (Monoid w) => MTrans (W.WriterT w) (W.Writer w) where
+    mtrans = id
+
+instance (Monoid w) => MTrans (W'.WriterT w) (W'.Writer w) where
+    mtrans = id
+
+instance (Monoid w) => MTrans (W.WriterT w) (((,) w)) where
```

Comment:
It looks like this instance causes issues on older versions of `base` according to CI, since they are missing a `Monad` instance for `(,) w`.  It might be necessary to guard this with CPP (i.e. `#ifdef MIN_VERSION_base(...)` or something like that)

Reply:
Have you got a travis set up so I can ensure my code supports everything it should?

Reply:
yes, yes you do :D